Sources/Misc/SystemInfo.swift

```diff
@@ -211,7 +212,30 @@ private extension SystemInfo {
         }
 
         guard let sharedUIApplication = self.sharedUIApplication else { return false }
-        return sharedUIApplication.applicationState == UIApplication.State.background
+        return sharedUIApplication.applicationState == .background
+    }
+
+    #elseif os(watchOS)
+
+    var isApplicationBackgroundedWatchOS: Bool {
+        // In Xcode 13 and earlier the system divides a watchOS app into two sections:
+        // - WatchKit app
+        // - WatchKit extension
+        // In Xcode 14 and later, you can produce watchOS apps with a single watchOS app target.
+        // These single-target watchOS apps can run on watchOS 7 and later.
+        // Calling `WKExtension.shared` on these single-target apps crashes.
+        //
+        // `WKApplication` provides a more accurate value if it's available, and it avoids that crash.
+        #if swift(>=5.7)
+        if #available(watchOS 7.0, *) {
+            return WKApplication.shared().applicationState == .background
+        } else {
+            return WKExtension.shared().applicationState == .background
+        }
+        #else
+        // Before Xcode 14, single-target extensions aren't supported (and WKApplication isn't available)
+        return WKExtension.shared().applicationState == .background
+        #endif
```

Comment:
Well this becomes slightly more complex, having to check for 2 things... But well, I don't have a better way to do it and it's not too bad.

Reply:
Yeah, but moving forward we'll only have one. Once Xcode 14 is mandatory, and when we drop anything before `watchOS 7.0`, there's really only one implementation :)